src/gui/src/components/ui/copy-to-clipboard.tsx

```diff
@@ -0,0 +1,92 @@
+import { motion, AnimatePresence } from 'framer-motion'
+import React from 'react'
+import { Copy, Check } from 'lucide-react'
+import { cva } from 'class-variance-authority'
+import type { VariantProps } from 'class-variance-authority'
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+  TooltipProvider,
+} from '@/components/ui/tooltip.jsx'
+import { cn } from '@/lib/utils.js'
+
+const copyToClipboardVariants = cva(
+  'inline-flex cursor-default transition-colors duration-150 gap-2 h-7 rounded-sm px-3 py-1.5 text-sm font-courier truncate bg-secondary text-muted-foreground items-center justify-center',
+)
+
+type CopyToClipboardProps = React.HTMLProps<HTMLButtonElement> &
+  VariantProps<typeof copyToClipboardVariants> & {
+    children?: React.ReactNode
+    copyValue?: string
+    toolTipText?: string
+  }
+
+const CopyToClipboard = React.forwardRef<
+  HTMLButtonElement,
+  CopyToClipboardProps
+>(({ children, className, copyValue, toolTipText }, ref) => {
+  const [copied, setCopied] = React.useState<boolean>(false)
+
+  const handleCopy = async () => {
+    if (!copyValue) return
+    try {
+      await navigator.clipboard.writeText(copyValue)
+      setCopied(true)
+      setTimeout(() => setCopied(false), 1500)
+    } catch (err) {
+      console.error('Failed to copy!', err)
```

Comment:
Wild idea, and def most an idea to think about, not something to change right now but I'm thinking that it would be a good idea to provide a fallback / graceful degradation here in this case, I'm not sure if it's going to be practical everywhere layout-wise but it would be nice to print the value in the html in case the copy failed, otherwise the user is never retrieving, let's say, the tarball value from the "Tarball" button in case that for some reason the clipboard API throws.